src/arkode/arkode_lsrkstep.c

```diff
@@ -0,0 +1,1436 @@
+                          rat * (rn * (rn + ONE) / TWO - ONE) * ark_mem->h,
+                        ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+  if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+  N_VLinearSum(rn / (TWO * rn - ONE), ark_mem->tempv2,
+               (rn - ONE) / (TWO * rn - ONE), ark_mem->ycur, ark_mem->ycur);
+  N_VLinearSum(ONE, ark_mem->ycur,
+               (rn - ONE) * rat * ark_mem->h / (TWO * rn - ONE), ark_mem->fn,
+               ark_mem->ycur);
+  N_VLinearSum(ONE, ark_mem->tempv1, ark_mem->h / rs, ark_mem->fn,
+               ark_mem->tempv1);
+
+  for (int j = (int)(in * (in + 1) / 2 + 1); j <= step_mem->reqstages; j++)
+  {
+    retval = step_mem->fe(ark_mem->tcur +
+                            ((sunrealtype)j - rn - ONE) * rat * ark_mem->h,
+                          ark_mem->ycur, ark_mem->fn, ark_mem->user_data);
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+
+    N_VLinearSum(ONE, ark_mem->ycur, ark_mem->h * rat, ark_mem->fn,
+                 ark_mem->ycur);
+    N_VLinearSum(ONE, ark_mem->tempv1, ark_mem->h / rs, ark_mem->fn,
+                 ark_mem->tempv1);
+  }
+
+  step_mem->nfe += step_mem->reqstages - 1;
+
+  /* Compute yerr (if step adaptivity enabled) */
+  if (!ark_mem->fixedstep)
+  {
+    N_VLinearSum(ONE, ark_mem->ycur, -ONE, ark_mem->tempv1, ark_mem->tempv1);
+
+    *dsmPtr = N_VWrmsNorm(ark_mem->tempv1, ark_mem->ewt);
+  }
+  if (*dsmPtr <= ONE || ark_mem->fixedstep)
+  {
+    retval = step_mem->fe(ark_mem->tcur + ark_mem->h, ark_mem->ycur,
+                          ark_mem->fn, ark_mem->user_data);
+    step_mem->nfe++;
+    ark_mem->fn_is_current = SUNTRUE;
+    if (retval != ARK_SUCCESS) { return (ARK_RHSFUNC_FAIL); }
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  lsrkStep_TakeStepSSP104:
+
+  This routine serves the primary purpose of the LSRKStepSSP104 module:
+  it performs a single SSP104 step (with embedding).
+
+  The output variable dsmPtr should contain estimate of the
+  weighted local error if an embedding is present; otherwise it
+  should be 0.
+
+  The input/output variable nflagPtr is used to gauge convergence
+  of any algebraic solvers within the step.  As this routine
+  involves no algebraic solve, it is set to 0 (success).
+
+  The return value from this routine is:
+            0 => step completed successfully
+           >0 => step encountered recoverable failure;
+                 reduce step and retry (if possible)
+           <0 => step encountered unrecoverable failure
+  ---------------------------------------------------------------*/
+int lsrkStep_TakeStepSSP104(ARKodeMem ark_mem, sunrealtype* dsmPtr, int* nflagPtr)
+{
+  int retval;
+
+  ARKodeLSRKStepMem step_mem;
+
+  /* initialize algebraic solver convergence flag to success,
+     temporal error estimate to zero */
+  *nflagPtr = ARK_SUCCESS;
+  *dsmPtr   = ZERO;
+
+  sunrealtype onesixth = 1.0 / 6.0;
```

Comment:
Use `SUN_RCONST` for both of these constants.
```suggestion
  const sunrealtype onesixth = ONE / SUN_RCONST(6.0);
```